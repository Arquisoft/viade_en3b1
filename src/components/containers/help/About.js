import React from 'react';
import NavBar from '../../ui/main/NavBar';
import Footer from '../../ui/main/Footer';
import { Container, Typography, makeStyles, Grid, Box, Link } from '@material-ui/core';
import svgIconMap from '../../../assets/img/icons/AboutMap.svg';
import svgIconInterop from '../../../assets/img/icons/AboutInterop.svg';
import svgIconMe from '../../../assets/img/icons/AboutMe.svg';
import svgIconViaDe from '../../../assets/img/icons/AboutViaDe.svg';
import svgIconData from '../../../assets/img/icons/AboutData.svg';

export default function About() {

    const classes = useStyles();

    return (
        <div>
            <NavBar />

            <Grid
                item
                xs={false}
                sm={12}
                md={12}
                className={classes.image}>
                <Typography align="center" className={classes.text}>ViaDe En3B1</Typography>
            </Grid>
            <Container
                component="main"
                className={classes.main}
                maxWidth="md"
            >
                <Grid container spacing={10}>
                    <Grid item xs={6}>
                        <img alt="map about" src={svgIconViaDe} style={{ height: '27rem', width: '27rem' }} />
                    </Grid>
                    <Grid item xs={6} className={classes.point}>
                        <Box fontSize="h4.fontSize" fontWeight="fontWeightBold">What is ViaDe?</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">It is the first routes management system based on SOLID technology. It has been developed as part of the <Link style={{ color: "#27DEBF" }} target="_blank" href={"https://arquisoft.github.io/"}>Software Architecture</Link> course at UniOvi.</Box>
                    </Grid>
                </Grid>

                <Grid container spacing={10}>
                    <Grid item xs={6} className={classes.point}>
                        <Box fontSize="h4.fontSize" fontWeight="fontWeightBold">Do you need to get somewhere?</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">With ViaDe, you can create or import any route you want so you can store them and check on them at any moment. You can learn more <Link style={{ color: "#27DEBF" }} href={"#/help"}>here</Link>.</Box>
                    </Grid>
                    <Grid item xs={5}>
                        <img alt="map about" src={svgIconMap} style={{ height: '27rem', width: '27rem' }} />
                    </Grid>
                </Grid>

                <Grid container spacing={10}>
                    <Grid item xs={6}>
                        <img alt="map about" src={svgIconData} style={{ height: '27rem', width: '27rem' }} />
                    </Grid>
                    <Grid item xs={6} className={classes.point}>
                        <Box fontSize="h4.fontSize" fontWeight="fontWeightBold">Your data? Easy.</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">All your routes are stored in your personal Pod. Also, we don't keep any personal information, so you can be calm about your data.</Box>
                    </Grid>
                </Grid>

                <Grid container spacing={10}>
                    <Grid item xs={6} className={classes.point}>
                        <Box fontSize="h4.fontSize" fontWeight="fontWeightBold">Interoperability</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">ViaDe En3B1 is interoperable with other ViaDe apps. This means that you can check your routes in other applications, as well as creating new ones. You can try it, it's awesome!</Box>
                    </Grid>
                    <Grid item xs={5}>
                        <img alt="map about" src={svgIconInterop} style={{ height: '27rem', width: '27rem' }} />
                    </Grid>
                </Grid>

                <Grid container spacing={10}>
                    <Grid item xs={6}>
                        <img alt="map about" src={svgIconMe} style={{ height: '27rem', width: '27rem' }} />
                    </Grid>
                    <Grid item xs={6} className={classes.point}>
                        <Box fontSize="h4.fontSize" fontWeight="fontWeightBold">Hi! I'm Pablo.</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">I am the developer behind ViaDe En3B1.</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">At the moment, I am a Software Engineering student at UniOvi. When I'm not programming, I enjoy photography, running and movies.</Box>
                        <Box fontSize="h6.fontSize" fontWeight="fontWeightRegular">You can contact me on <Link target="_blank" style={{ color: "#27DEBF" }} href={"https://www.linkedin.com/in/pablocanalsuarez/"}>LinkedIn</Link>, <Link style={{ color: "#27DEBF" }} target="_blank" href={"https://github.com/PabloCanalSuarez"}>GitHub</Link> or by <Link style={{ color: "#27DEBF" }} href={"mailto:uo264903@uniovi.es"}>Email</Link>.</Box>
                    </Grid>
                </Grid>


            </Container>
            <Footer />
        </div>
    )
}

const useStyles = makeStyles((theme) => ({
    main: {
        marginTop: theme.spacing(8),
        marginBottom: theme.spacing(8),
    },
    text: {
        fontSize: '12vw',
        color: '#000000'
        // color: '#ffffff'
    },
    image: {
        // backgroundImage: `url(https://source.unsplash.com/collection/9847024/1600x900)`,
        backgroundRepeat: 'no-repeat',
        // backgroundColor: '#4EFFBB',
        backgroundColor: '#27DEBF',
        backgroundSize: 'cover',
        backgroundPosition: 'center',
        height: '20vw',
        // minHeight: '90vh',
    },
    point: {
        marginTop: theme.spacing(15)
    }
}));