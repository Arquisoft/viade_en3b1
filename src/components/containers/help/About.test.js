import About from "./About";

test("About", () => {
    expect(About).toBeDefined();
});

test("Renders Correct", () => {
    const wrapper = render(<About />);
    expect(wrapper).toMatchSnapshot();
});