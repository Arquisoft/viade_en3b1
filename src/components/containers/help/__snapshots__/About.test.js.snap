// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correct 1`] = `
<div>
  <div
    class="makeStyles-root-5"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-6 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root makeStyles-link-8"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
              href="#/home"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    alt="Viade Logo"
                    src="logo_2.svg"
                    style="height:2.5rem;width:2.5rem"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    ViaDe
                  </h4>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    class="MuiGrid-root makeStyles-image-3 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
  >
    <p
      class="MuiTypography-root makeStyles-text-2 MuiTypography-body1 MuiTypography-alignCenter"
    >
      ViaDe En3B1
    </p>
  </div>
  <main
    class="MuiContainer-root makeStyles-main-1 MuiContainer-maxWidthMd"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <img
          alt="map about"
          src="AboutViaDe.svg"
          style="height:27rem;width:27rem"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-point-4 MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiBox-root MuiBox-root-200"
        >
          What is ViaDe?
        </div>
        <div
          class="MuiBox-root MuiBox-root-201"
        >
          It is the first routes management system based on SOLID technology. It has been developed as part of the 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://arquisoft.github.io/"
            style="color:#27DEBF"
            target="_blank"
          >
            Software Architecture
          </a>
           course at UniOvi.
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        class="MuiGrid-root makeStyles-point-4 MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiBox-root MuiBox-root-202"
        >
          Do you need to get somewhere?
        </div>
        <div
          class="MuiBox-root MuiBox-root-203"
        >
          With ViaDe, you can create or import any route you want so you can store them and check on them at any moment. You can learn more 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="#/help"
            style="color:#27DEBF"
          >
            here
          </a>
          .
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <img
          alt="map about"
          src="AboutMap.svg"
          style="height:27rem;width:27rem"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <img
          alt="map about"
          src="AboutData.svg"
          style="height:27rem;width:27rem"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-point-4 MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiBox-root MuiBox-root-204"
        >
          Your data? Easy.
        </div>
        <div
          class="MuiBox-root MuiBox-root-205"
        >
          All your routes are stored in your personal Pod. Also, we don't keep any personal information, so you can be calm about your data.
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        class="MuiGrid-root makeStyles-point-4 MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiBox-root MuiBox-root-206"
        >
          Interoperability
        </div>
        <div
          class="MuiBox-root MuiBox-root-207"
        >
          ViaDe En3B1 is interoperable with other ViaDe apps. This means that you can check your routes in other applications, as well as creating new ones. You can try it, it's awesome!
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <img
          alt="map about"
          src="AboutInterop.svg"
          style="height:27rem;width:27rem"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <img
          alt="map about"
          src="AboutMe.svg"
          style="height:27rem;width:27rem"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-point-4 MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiBox-root MuiBox-root-208"
        >
          Hi! I'm Pablo.
        </div>
        <div
          class="MuiBox-root MuiBox-root-209"
        >
          I am the developer behind ViaDe En3B1.
        </div>
        <div
          class="MuiBox-root MuiBox-root-210"
        >
          At the moment, I am a Software Engineering student at UniOvi. When I'm not programming, I enjoy photography, running and movies.
        </div>
        <div
          class="MuiBox-root MuiBox-root-211"
        >
          You can contact me on 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://www.linkedin.com/in/pablocanalsuarez/"
            style="color:#27DEBF"
            target="_blank"
          >
            LinkedIn
          </a>
          , 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://github.com/PabloCanalSuarez"
            style="color:#27DEBF"
            target="_blank"
          >
            GitHub
          </a>
           or by 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="mailto:uo264903@uniovi.es"
            style="color:#27DEBF"
          >
            Email
          </a>
          .
        </div>
      </div>
    </div>
  </main>
  <div>
    <footer
      class="makeStyles-footer-214"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              ViaDe En3B1
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Made by 
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://www.linkedin.com/in/pablocanalsuarez/"
                style="color:#302c58"
                target="_blank"
              >
                Pablo Ca√±al
              </a>
            </h6>
            <div>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://www.linkedin.com/in/pablocanalsuarez/"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="mailto:uo264903@uniovi.es"
                style="margin-left:0.5rem"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </a>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
          />
          <div
            class="MuiGrid-root MuiGrid-item"
            style="margin-top:1rem"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://github.com/Arquisoft/viade_en3b1/"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
